client/src/main/java/org/eclipse/hono/client/impl/ApplicationClientFactoryImpl.java

```diff
@@ -167,6 +171,7 @@ private void removeCommandClient(final String key) {
                     () -> AsyncCommandClientImpl.create(
                             connection,
                             tenantId,
+                            samplerFactory.create(CommandConstants.NORTHBOUND_COMMAND_REQUEST_ENDPOINT),
```

Comment:
is there a particular reason why you only want to add the sampler to the async command client and not the one returned by *getOrCreateCommandClient()*? In general, I wonder if an northbound application client should be required to provide a sampler factory when creating the ApplicationClientFactory. IMHO we should at least also provide a constructor that doesn't require a sampler factory but instead simply uses the Noop one under the hood. WDYT?